modules/cactus-web/src/FileInput/FileInput.tsx

```diff
@@ -0,0 +1,624 @@
+      top: 14px;
+      color: ${p => p.theme.colors.white};
+    }
+
+    ${ActionsRefresh} {
+      height: 12px;
+      width: 12px;
+    }
+  }
+`
+
+const defaultErrorHandler = (errorType: ErrorType, accept: string[] | undefined = []): string => {
+  let errorMsg = 'An unknown error occurred when reading the file.'
+  switch (errorType) {
+    case FILE_TYPE_ERR:
+      errorMsg = `The file provided does not match any of the accepted file types: ${accept.join(
+        ', '
+      )}`
+      break
+    case NOT_FOUND_ERR:
+      errorMsg = 'The file provided could not be found. Please try again.'
+      break
+    case SECURITY_ERR:
+      errorMsg = 'The file could not be read due to security restrictions.'
+      break
+    case ABORT_ERR:
+      errorMsg = 'The file read operation was aborted. Please try again.'
+      break
+    case NOT_READABLE_ERR:
+      errorMsg =
+        'The file read operation failed. The file may be too large. Please try again or select a different file.'
+      break
+    case ENCODING_ERR:
+      errorMsg = 'The encoding or decoding operation failed. Please try again.'
+      break
+  }
+
+  return errorMsg
+}
+
+const FileInputBase = (props: FileInputProps) => {
+  const {
+    className,
+    onChange,
+    onError = defaultErrorHandler,
+    rawFiles,
+    name,
+    accept,
+    labels,
+    multiple,
+    value,
+    ...fileInputProps
+  } = props
+  const [state, setState] = useState<State>({
+    files: value || [],
+    inputKey: Math.random().toString(36),
+  })
+  const isMounted = useRef(true)
+  const fileSelector = useRef<HTMLInputElement | null>(null)
+  const topFileBox = useRef<HTMLDivElement | null>(null)
+
+  useEffect(() => {
+    return () => {
+      isMounted.current = false
+    }
+  }, [])
+
+  useEffect(() => {
+    if (topFileBox.current) {
+      topFileBox.current.focus()
+    }
+  }, [state.files])
+
+  const saveFiles = (files: FileList | null) => {
+    if (files && files.length > 0) {
+      const loadingPromises = Array.from(files).map(
+        file =>
+          new Promise<FileObject>(resolve => {
+            resolve({ fileName: file.name, contents: null, status: 'loading' })
+          })
+      )
+
+      Promise.all(loadingPromises).then(results => {
+        if (isMounted.current === true) {
+          setState(state => ({ ...state, files: results }))
+          if (typeof onChange === 'function') {
+            onChange(name, results)
+          }
+        }
+      })
+
+      const readingPromises = Array.from(files).map(file => {
+        if (!accepts(file, accept)) {
+          return new Promise<FileObject>(resolve => {
+            const errorMsg = onError(FILE_TYPE_ERR, accept)
+            resolve({
+              fileName: file.name,
+              contents: null,
+              status: 'error',
+              errorMsg: errorMsg,
+            })
+          })
+        }
+
+        if (rawFiles) {
+          return new Promise<FileObject>(resolve => {
```

Comment:
```suggestion
          return Promise.resolve({
            fileName: file.name,
            contents: file,
            status: 'loaded'
          })
```
You can actually just use Promise.resolve here, and it's the same just a little cleaner.